rest-api/openapi/custom.yaml

```diff
@@ -0,0 +1,317 @@
+# Copyright DB InfraGO AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+openapi: 3.0.3
+info:
+  title: Capella API
+  description: API to access live data and modify Capella projects
+  version: 0.0.1
+servers:
+  - url: http://localhost:5007/api/v1
+    description: Embedded Capella REST API server
+tags:
+  - name: Diagrams
+    description: >
+      Operations related to diagrams in a Capella model
+  - name: Logical Architectures
+    description: >
+      Operations related to logical architecture layers in a Capella model
+  - name: Projects
+    description: >
+      Operations related to projects in the current workspace in Capella
+paths:
+  /projects:
+    get:
+      tags:
+        - Projects
+      summary: List all projects in the workspace
+      operationId: listProjects
+      responses:
+        '200':
+          description: A list of projects
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/WorkspaceProject'
+    post:
+      tags:
+        - Projects
+      summary: Import a project into the workspace
+      operationId: importProject
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - location
+              properties:
+                location:
+                  type: string
+                  description: >
+                    Absolute path of the project folder in the local file
+                    system
+      responses:
+        '201':
+          description: Project imported successfully.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/WorkspaceProject'
+        '400':
+          description: Invalid request
+        '404':
+          description: Project not found
+        '409':
+          description: Project already exists in the workspace
+  /projects/{project_name}:
+    get:
+      tags:
+        - Projects
+      summary: Get a project by name
+      operationId: getProjectByName
+      parameters:
+        - name: project_name
+          in: path
+          required: true
+          description: Unique name of the project
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/WorkspaceProject'
+        '404':
+          description: Project not found
```

Comment:
I suppose this is the definition for the unimplemented `getProjectByName()` function, which should then also be disappearus-ed, in order to not confuse downstream consumers.

Reply:
done